include/aspect/material_model/equation_of_state/linearized_incompressible.h

```diff
@@ -0,0 +1,91 @@
+/*
+  Copyright (C) 2011 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_material_model_equation_of_state_linearized_incompressible_h
+#define _aspect_material_model_equation_of_state_linearized_incompressible_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/material_model/equation_of_state/interface.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace EquationOfState
+    {
+      using namespace dealii;
+
+      /**
+       * A simplified, incompressible equation of state where the density depends linearly
+       * on temperature and composition, using the equation
+       * $\rho(p,T,\mathfrak c) = \left(1-\alpha (T-T_0)\right)\rho_0 + \sum_i \Delta\rho_i \; \mathfrak c_i.$ "
+       * There is no pressure-dependence of the density, and all other material properties
+       * relating to the equation of state are assumed to be constant and identical for each
+       * composition.
+       */
+      template <int dim>
+      class LinearizedIncompressible
+      {
+        public:
+          void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
```

Comment:
Add a short documentation. `Function that computes the output of the equation of state for all compositions` should be sufficient.

Reply:
I don't think I understand what `q` is supposed to be. You say "which entry of the vector of inputs is used", but the input is a structure and not an array. Can you clarify?